From cf29dcd37cde8e991f19642b8cbf27a95f2ac958 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Wed, 6 Apr 2022 18:18:55 +0200
Subject: [PATCH 160/841] refactor(stm32mp2-fdts): move on IOM bindings

This patch adds IOM node and moves OSPI nodes in IOM node.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 8cc14cac786a0afd2ec52ff87c0691bbb12f8482 refactor(stm32mp2-fdts): move on IOM bindings]
Change-Id: I2aee5f17f577dffac737d2e71bd4ad427b4271e7
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/248423
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
---
 fdts/stm32mp25-bl2.dtsi |  7 +++++--
 fdts/stm32mp251.dtsi    | 44 ++++++++++++++++++++++++-----------------
 2 files changed, 31 insertions(+), 20 deletions(-)

diff --git a/fdts/stm32mp25-bl2.dtsi b/fdts/stm32mp25-bl2.dtsi
index 6885e480a..c1f8b4a3e 100644
--- a/fdts/stm32mp25-bl2.dtsi
+++ b/fdts/stm32mp25-bl2.dtsi
@@ -6,8 +6,11 @@
 / {
 	soc@0 {
 #if !STM32MP_SPI_NAND && !STM32MP_SPI_NOR
-		/delete-node/ spi@40430000;
-		/delete-node/ spi@40440000;
+		/delete-node/ ommanager@40500000;
+#else
+		ommanager@40500000 {
+			/delete-node/ spi@40440000;
+		};
 #endif
 #if !STM32MP_RAW_NAND
 		/delete-node/ memory-controller@48200000;
diff --git a/fdts/stm32mp251.dtsi b/fdts/stm32mp251.dtsi
index 2c4f473d3..016743c40 100644
--- a/fdts/stm32mp251.dtsi
+++ b/fdts/stm32mp251.dtsi
@@ -99,26 +99,34 @@
 			};
 		};
 
-		ospi1: spi@40430000 {
-			compatible = "st,stm32-omi";
-			reg = <0x40430000 0x400>, <0x60000000 0x8000000>;
-			reg-names = "ospi", "ospi_mm";
-			interrupts = <GIC_SPI 163 IRQ_TYPE_LEVEL_HIGH>;
-			st,syscfg-dlyb = <&syscfg 0x1000>;
-			clocks = <&rcc CK_KER_OSPI1>;
-			resets = <&rcc OSPI1_R>, <&rcc OSPI1DLL_R>;
+		ommanager: ommanager@40500000 {
+			#address-cells = <2>;
+			#size-cells = <1>;
+			compatible = "st,stm32mp25-omm";
+			reg = <0x40500000 0x400>, <0x60000000 0x10000000>;
+			reg-names = "omm", "omm_mm";
+			clocks = <&rcc CK_BUS_OSPIIOM>;
+			resets = <&rcc OSPIIOM_R>;
 			status = "disabled";
-		};
 
-		ospi2: spi@40440000 {
-			compatible = "st,stm32-omi";
-			reg = <0x40440000 0x400>, <0x68000000 0x8000000>;
-			reg-names = "ospi", "ospi_mm";
-			interrupts = <GIC_SPI 164 IRQ_TYPE_LEVEL_HIGH>;
-			st,syscfg-dlyb = <&syscfg 0x1400>;
-			clocks = <&rcc CK_KER_OSPI2>;
-			resets = <&rcc OSPI2_R>, <&rcc OSPI2DLL_R>;
-			status = "disabled";
+			ranges = <0 0 0x40430000 0x400>,
+				 <1 0 0x40440000 0x400>;
+
+			ospi1: spi@40430000 {
+				compatible = "st,stm32mp25-omi";
+				reg = <0 0 0x400>;
+				clocks = <&rcc CK_KER_OSPI1>;
+				resets = <&rcc OSPI1_R>, <&rcc OSPI1DLL_R>;
+				status = "disabled";
+			};
+
+			ospi2: spi@40440000 {
+				compatible = "st,stm32mp25-omi";
+				reg = <1 0 0x400>;
+				clocks = <&rcc CK_KER_OSPI2>;
+				resets = <&rcc OSPI2_R>, <&rcc OSPI2DLL_R>;
+				status = "disabled";
+			};
 		};
 
 		rng: rng@42020000 {
-- 
2.47.2

